helm-v3/006-repositories.md

```diff
@@ -1,13 +1,28 @@
 # Repositories
 
+A new API version for chart repositories (**v2**) will be introduced to provide new functionality address issues of scale (see [#3557](https://github.com/kubernetes/helm/issues/3557)).
+
 The following changes will be made to the repository subsystem:
 
-- The `helm serve` command will be removed
+- v2 will be the new default
+- v1 will be supported in its current state, but will not benefit from new functionality
+- The `helm search` command will become a server-side operation for v2 repos
```

Comment:
To add context...

This will preclude object storage from being using with the v2 specification. To maintainer object storage, and other flat system, compatibility we'd need to keep v1 around indefinitely.

The scaling issue in [#3557](https://github.com/kubernetes/helm/issues/3557) is an issue to the extent it is because of the YAML parser we use and how it works. There is a proposal on the issue that would still work for YAML index files. Another option would be to write a YAML parser that didn't use as much memory.

Prior to making this move I'd like to have a better handle on what the community needs. To do that I'm proposing a Helm survey (details coming soon) to ask people. Until then I'd like to see this item be an open question we move to solve quickly.

Reply:
The ideal solution will be to do both i.e. improve the yaml parser and move towards nested indexes.